plugin-mainmenu/lxqtmainmenu.cpp

```diff
@@ -96,7 +100,9 @@ LXQtMainMenu::LXQtMainMenu(const ILXQtPanelPluginStartupInfo &startupInfo):
     mSearchEdit = new QLineEdit;
     mSearchEdit->setClearButtonEnabled(true);
     mSearchEdit->setPlaceholderText(LXQtMainMenu::tr("Search..."));
-    connect(mSearchEdit, &QLineEdit::textChanged, this, &LXQtMainMenu::searchTextChanged);
+    connect(mSearchEdit, &QLineEdit::textChanged, [this] (QString const &) {
```

Comment:
why the extra lambda here? isn't `connect(mSearchEdit, &QLineEdit::textChanged, &mSearchTimer, static_cast<void (QTimer::*)()>(&QTimer::start));` enough (and more efficient)?

Reply:
Lambda feels cleaner than `static_cast<void (QTimer::*)()>` ;) As for efficiency, I don't think there's a real difference.

Reply:
> Lambda feels cleaner than static_cast<void (QTimer::*)()> 

Matter of taste, for me avoiding unneeded lambda is definitely a better choice.

> As for efficiency, I don't think there's a real difference.

Billions of drops will form an ocean. Ignoring little inefficiencies (each at a time) can lead you/us to QML :-)

Reply:
> ... can lead you/us to QML :-)

LOL! You know how I love those three letters ;)

> Matter of taste, for me avoiding unneeded lambda is definitely a better choice.

Sometimes they make the code (much) more readable while keeping it compact. I agree with you that it's a matter of taste. I think I neither avoid nor overuse them.

And, years ago, I learned from you that the context of the connection can't be overestimated (although, in this case, it wasn't needed).